lib/compat/wordpress-6.0/block-patterns.php

```diff
@@ -39,3 +39,92 @@ function gutenberg_register_remote_theme_patterns() {
 }
 
 add_action( 'init', 'gutenberg_register_remote_theme_patterns' );
+
+/**
+ * Register any patterns that the active theme may provide under its
+ * `./patterns/` directory. Each pattern is defined as a PHP file and defines
+ * its metadata using plugin-style headers. The minimum required definition is:
+ *
+ *     /**
+ *      * Pattern Name: My Pattern
+ *      *
+ *
+ * The output of the PHP source corresponds to the content of the pattern, e.g.:
+ *
+ *     <main><p><?php echo "Hello"; ?></p></main>
+ *
+ * If applicable, this will collect from both parent and child theme.
+ *
+ * @since 6.0.0
+ * @access private
+ * @internal
+ */
+function gutenberg_register_theme_block_patterns() {
+	$default_headers = array(
+		'title'         => 'Pattern Name',
+		'description'   => 'Description',
+		'viewportWidth' => 'Viewport Width',
+		'categories'    => 'Categories',
+		'keywords'      => 'Keywords',
+		'blockTypes'    => 'Block Types',
+	);
+
+	// Register patterns for the active theme, for both parent and child theme,
+	// if applicable.
+	foreach ( wp_get_active_and_valid_themes() as $theme ) {
```

Comment:
I have just realized that this hook will run for every page load. It's concerning for the front end where every additional processing matters. Should we limit dynamic loading patterns only to pages/screens where we know that the block editor is going to be present?

Reply:
Huh, good point. 🤦 Using `'init'` was OK for the purpose of showing the concept of the PR, and now the code has been sitting there for so long that it's hard to notice these little things.